### PR TITLE
Rename duplicate files when uploading

### LINKED ISSUE #2987: Uploading a file with the same name overwrites the old one
## Steps to reproduce
1. Upload `Readme.md` in conversation A
2. Upload another different `Readme.md` in conversation B
3. Realise the `Readme.md` in conversation A is updated.

### Expected behaviour
In B a new file `Readme (2).md` should be uploaded

### Actual behaviour
Update of the existing file

### Solution
The android app sends a HEAD request to the webdav endpoint to check if the file already exists. If it does, `(2)` is added before the last `.` in the filename (or at the end in case there is no dot). Then send another head request. In case of another match, search for ` \(\d+\)` before the last dot, and increase the number by 1. Repeat this until the file files not exist.

### PR DESCRIPTION
**Pull request by ma12-co**
fix #2987 

Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

File: src/store/fileUploadStore.js [hunk 1]
```diff
@@ -108,44 +112,36 @@ const mutations = {
...
 			// Destination path on the server
 			const path = '/files/' + userId + getters.getAttachmentFolder() + '/' + currentFile.name
+			const uniquePath = await findUniquePath(client, path)
 			try {
 				// Upload the file
-				await client.putFileContents(path, currentFile)
+				await client.putFileContents(uniquePath, currentFile)
+				// Compute the sharePath by removing the first part of the upload path
+				const partOfPathToRemove = '/files/' + userId
+				const sharePath = uniquePath.substring(partOfPathToRemove.length, uniquePath.length)
```
Comment (nickvergessen) [inline src/store/fileUploadStore.js]: make your life easier by not passing `'/files/' + userId` to `findUniquePath` in first place?

Comment (ma12-co) [reply src/store/fileUploadStore.js]: This has to be done, whether it's in findUniquePath or outside of it. So no easier life I think :)

Comment (nickvergessen) [reply src/store/fileUploadStore.js]: I ment something like https://gist.github.com/nickvergessen/e88c72d68fbc5d457a2c145c9e880954

**Comment by nickvergessen** [comment]
For me the sharing throws a 404, because the path is missing from the sent parameters:
```json
{"shareType":10,"shareWith":"vnia5iw6"}
```

**Pull request merged**